compile.go

```diff
@@ -47,54 +47,72 @@ func Compile(ctx context.Context, q string, now time.Time, opts ...Option) (*Spe
 	}
 
 	s, _ := opentracing.StartSpanFromContext(ctx, "parse")
-	itrp := NewInterpreter()
-	itrp.SetOption(nowOption, nowFunc(now))
-	if err := Eval(itrp, q); err != nil {
+
+	functionCalls, scope, err := Eval(q, SetOption(nowOption, nowFunc(now)))
+	if err != nil {
 		return nil, err
 	}
+
 	s.Finish()
 	s, _ = opentracing.StartSpanFromContext(ctx, "compile")
 	defer s.Finish()
 
-	spec := toSpecFromSideEffecs(itrp)
+	nowOpt, ok := scope.Lookup(nowOption)
+	if !ok {
+		return nil, fmt.Errorf("%q option not set", nowOption)
+	}
+
+	nowTime, err := nowOpt.Function().Call(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := ToSpec(functionCalls, nowTime.Time().Time())
+	if err != nil {
+		return nil, err
+	}
 
 	if o.verbose {
 		log.Println("Query Spec: ", Formatted(spec, FmtJSON))
 	}
 	return spec, nil
 }
 
-// Eval evaluates the flux string q and update the given interpreter
-func Eval(itrp *interpreter.Interpreter, q string) error {
-	astPkg := parser.ParseSource(q)
+func Eval(flux string, opts ...scopeMutator) ([]values.Value, interpreter.Scope, error) {
```

Comment:
nit, can you make `scopeMutator` public? Since it it part of the public API here.